pkg/internal/gcp-get-hostname.ps1

```diff
@@ -0,0 +1,33 @@
+# This script returns a mutated hostname if the length is longer than 63 characters for GCP. The hostname will be
+# the lesser of 63 characters after the first dot in the FQDN.
+#
+# To get the original hostname for a GCP compute instance, the instance metadata service is invoked using it's
+# IPv4 address as the prefered method over the DNS address or other alternatives like the GoogleCloud Tools for
+# PowerShell library (https://cloud.google.com/tools/powershell/docs/quickstart) since they may not be available,
+# specially in the BYOH use case.
+#
+# The IPv4 address of the instance metadata service is defaulted to 169.254.169.254. See instance metadata
+# documentaion for GCP https://cloud.google.com/compute/docs/metadata/default-metadata-values.
+
+# MAX_LENGTH is the maximum number of character allowed for the instance's hostname in GCP
+$MAX_LENGTH = 63
+
+# get hostname from the instance metadata service
+$hostname=(Invoke-RestMethod -Headers @{'Metadata-Flavor'='Google'} `
+            -Uri "http://169.254.169.254/computeMetadata/v1/instance/hostname")
```

Comment:
Going with the doc this should be http://metadata.google.internal and not the IP address.

Reply:
There is a [comment](https://github.com/openshift/windows-machine-config-operator/pull/1284/files#diff-e5bc5d4c1a111c9f29a32845d5d34069f73857555f955fe898efeed077cdcba7R4) up in this file with details about this. 

TL;DR
Using the IP address is safer, especially in the BYOH use case where there is no control over the host file and the resolution of the  `metadata.google.internal` DNS name may not be available.

Reply:
Won't images be created from existing master images which will have this DNS setup? Now we could run into issues if the IP has changed. Or is there some way to import images created on prem where this need not have setup?